sw/device/tests/BUILD

```diff
@@ -1108,6 +1110,65 @@ opentitan_test(
     ],
 )
 
+# hash table for lc_state name and value.
+# 'name' is picked for pre generated otp image
+# 'value' is copied from
+#   https://cs.opensource.google/opentitan/opentitan/
+#   +/master:rules/const.bzl;drc=30f44025dfb4bc2bd328a99a0a810c3189fb2544;l=29
+_FLASH_CTRL_INFO_ACCESS_LC_STATES = {
+    "test_unlocked0": 0x02108421,
+    "dev": 0x21084210,
+    "prod": 0x2318c631,
+    "prod_end": 0x25294a52
+}
+
+load(
+    "//rules:opentitan.bzl",
+    "RSA_ONLY_KEY_STRUCTS",
+)
+load(
+    "//rules/opentitan:defs.bzl",
+    "rsa_key_for_lc_state",
+)
+
+# This loop will create following sival bazel target
+#  flash_ctrl_info_access_lc_test_unlocked0
+#  flash_ctrl_info_access_lc_test_dev
+#  flash_ctrl_info_access_lc_test_prod
+#  flash_ctrl_info_access_lc_test_prod_end
+
+[
+    opentitan_test(
+        name = "flash_ctrl_info_access_lc_{}".format(lc_state),
+        srcs = ["flash_ctrl_info_access_lc.c"],
+        cw310 = new_cw310_params(
+            bitstream = "//hw/bitstream:rom_with_fake_keys_otp_{}".format(lc_state),
+        ),
+        exec_env = {
+            "//hw/top_earlgrey:fpga_cw310_rom_with_fake_keys": None,
+        },
+        rsa_key = rsa_key_for_lc_state(
+            RSA_ONLY_KEY_STRUCTS,
+            _FLASH_CTRL_INFO_ACCESS_LC_STATES[lc_state],
+        ),
+        deps = [
+            "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+            "//sw/device/lib/base:mmio",
+            "//sw/device/lib/dif:flash_ctrl",
+            "//sw/device/lib/dif:lc_ctrl",
+            "//sw/device/lib/dif:rv_plic",
+            "//sw/device/lib/runtime:irq",
+            "//sw/device/lib/runtime:log",
+            "//sw/device/lib/testing:flash_ctrl_testutils",
+            "//sw/device/lib/testing:isr_testutils",
+            "//sw/device/lib/testing:lc_ctrl_testutils",
+            "//sw/device/lib/testing:rv_plic_testutils",
+            "//sw/device/lib/testing/test_framework:ottf_main",
+        ],
+    )
+    for lc_state, lc_state_val in _FLASH_CTRL_INFO_ACCESS_LC_STATES.items()
```

Comment:
This can be updated to the following in addition to the comment above regarding how to use the helper functions included in const.bzl.

```python
   for lc_state, lc_state_val in _FLASH_CTRL_INFO_ACCESS_LC_STATES
```

Reply:
updated